CRM/Report/Form/Contribute/PCP.php

```diff
@@ -190,6 +211,16 @@ public function from() {
 LEFT JOIN civicrm_contribution_page {$this->_aliases['civicrm_contribution_page']}
           ON {$this->_aliases['civicrm_pcp']}.page_id =
              {$this->_aliases['civicrm_contribution_page']}.id";
+
+    // for credit card type
+    if ($this->isTableSelected('civicrm_financial_trxn')) {
```

Comment:
I'm still leaning towards having a function for all these join additions since it's the same code going into multiple reports